File: 98.UVa.old/11292.py
while True:
  a = input().split()
  n = int(a[0])
  m = int(a[1])
  if n==0 and m==0:
    break
  r = []
  h = []
  for i in range(n):
    r.append(int(input()))
  for i in range(m):
    h.append(int(input()))
  r.sort()
  h.sort()

  i = j = ans = 0
  while i<n and j<m:
    if r[i]<=h[j]:
      ans += h[j]
      i+=1
    j+=1

  if i<n:
    print("Loowater is doomed!")
  else:
    print(ans)

File: 50.c/09.UVa/11292.c
#include<stdio.h>
#include<stdlib.h>

#define SWAP(x,y) {int Temp=(x);(x)=(y);(y)=Temp;}

int a[20000],b[20000];

void Sort(int a[], int left, int right);

int main(void){
	int n,m;

	while(scanf("%d%d",&n,&m)==2){
		if(n==0 && m==0) break;
		for(int i=0;i<n;i++) scanf("%d",&a[i]);
		for(int i=0;i<m;i++) scanf("%d",&b[i]);
		Sort(a,0,n-1);
		Sort(b,0,m-1);

		int cur=0,i=0,sum=0;
		while(cur<n && i<m){
			if(b[i++]>=a[cur]){
				sum+=b[i-1];
				cur++;
			}
		}
		if(cur<n) printf("Loowater is doomed!\n");
		else printf("%d\n",sum);
	}

	return 0;
}

void Sort(int a[], int left, int right){
	if(left>=right) return;
	int i=left, j=right, k=a[left];
	while(i!=j){
		while(i<j && a[j]>=k) j--;
		while(i<j && a[i]<=k) i++;
		SWAP(a[j],a[i]);
	}
	SWAP(a[left],a[i]);
	Sort(a,left,i-1);
	Sort(a,i+1,right);
}
